grails-app/domain/org/pih/warehouse/core/Location.groovy

```diff
@@ -95,6 +95,24 @@ class Location implements Comparable<Location>, java.io.Serializable {
         sortOrder(nullable: true)
     }
 
+    static uniqueNameValidator = { String name, Location obj ->
+        Location.withNewSession {
+            List<Location> otherLocations
+            if (obj.parentLocation) {
+                otherLocations = Location.findAllByNameAndParentLocation(name, obj.parentLocation)
+            } else {
+                otherLocations = Location.findAllByNameAndParentLocationIsNull(name)
+            }
+
+            // Exclude edited location from otherLocations (since it is validated withNewSession)
+            if (obj.id) {
```

Comment:
@jmiranda, @alannadolny I had to do this and then check `otherLocations?.size() > 0`, because:
- on create new (no other location with this name existing) `otherLocations.size()` was 0,
- on create new (1 other location with this name existing) `otherLocations.size()` was 1,
- on edit (no other location with this name existing) `otherLocations.size()` was 1,
- on edit (1 other location with this name existing) `otherLocations.size()` was 2,